File: benchmarks/poly/matrixmul/matrixmul.go
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	t0 := time.Now()
	n := 200
	a := make([][]float64, n)
	b := make([][]float64, n)
	for i := 0; i < n; i++ {
		a[i] = make([]float64, n)
		b[i] = make([]float64, n)
		for j := 0; j < n; j++ {
			a[i][j] = float64((i*n + j) % 13)
			b[i][j] = float64((j*n + i) % 17)
		}
	}

	c := make([][]float64, n)
	for i := 0; i < n; i++ {
		c[i] = make([]float64, n)
		for j := 0; j < n; j++ {
			s := 0.0
			for k := 0; k < n; k++ {
				s += a[i][k] * b[k][j]
			}
			c[i][j] = s
		}
	}

	checksum := 0.0
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			checksum += c[i][j]
		}
	}

	fmt.Println(int(checksum))
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}

File: benchmarks/poly/matrixmul/matrixmul.py
import sys, time


def main():
    _t0 = time.perf_counter_ns()
    n = 200
    a = [[float((i * n + j) % 13) for j in range(n)] for i in range(n)]
    b = [[float((j * n + i) % 17) for j in range(n)] for i in range(n)]
    c = [[0.0] * n for _ in range(n)]

    for i in range(n):
        for j in range(n):
            s = 0.0
            for k in range(n):
                s += a[i][k] * b[k][j]
            c[i][j] = s

    checksum = 0.0
    for i in range(n):
        for j in range(n):
            checksum += c[i][j]

    print(int(checksum))
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()
